### PR TITLE
Wallet api

### PR DESCRIPTION
**Pull request by Markuu-s**

### CODE REVIEW

File: core/api/full_node/node_api.hpp [hunk 1]
```diff
@@ -624,6 +624,12 @@ namespace fc::api {
                jwt::kReadPermission,
                MarketDealMap,
                const TipsetKey &)
+    API_METHOD(MarketAddBalance,
+               jwt::kSignPermission,
+               boost::optional<CID>,
```
Comment (turuslan) [inline core/api/full_node/node_api.hpp:629]:
```suggestion
               CID,
```

File: core/api/wallet/wallet_api.hpp [hunk 2]
```diff
@@ -38,6 +38,10 @@ namespace fc::api {
 
     API_METHOD(WalletSetDefault, jwt::kWritePermission, void, const Address &)
 
+    API_METHOD(WalletList, jwt::kAdminPermission, std::vector<Address>)
+
+    API_METHOD(WalletDelete, jwt::kReadPermission, void, const Address &)
```
Comment (turuslan) [inline core/api/wallet/wallet_api.hpp:43]:
```suggestion
    API_METHOD(WalletDelete, jwt::kAdminPermission, void, const Address &)
```

File: core/api/full_node/make.cpp [hunk 3]
```diff
@@ -747,6 +747,26 @@ namespace fc::api {
           }));
       return map;
     };
+    api->MarketAddBalance = [=](auto &address, auto &wallet, auto &amount)
+        -> outcome::result<boost::optional<CID>> {
```
Comment (turuslan) [inline core/api/full_node/make.cpp:751]:
```suggestion
        -> outcome::result<CID> {
```

File: core/api/full_node/node_api.hpp (hunk 1 above)
Comment (Markuu-s) [reply core/api/full_node/node_api.hpp]: fixed

File: core/api/wallet/wallet_api.hpp (hunk 2 above)
Comment (Markuu-s) [reply core/api/wallet/wallet_api.hpp]: fixed

File: core/api/full_node/make.cpp (hunk 3 above)
Comment (Markuu-s) [reply core/api/full_node/make.cpp]: fixed

File: core/api/wallet/local_wallet.cpp [hunk 4]
```diff
@@ -65,6 +66,27 @@ namespace fc::api {
...
+          seen.insert(address);
+        }
+      }
+
+      std::vector<Address> out;
+      while (not seen.empty()) {
+        out.emplace_back(std::move(seen.extract(seen.begin()).value()));
+      }
+
+      return out;
```
[the remark this answers is not in the document]
Comment (Markuu-s) [reply core/api/wallet/local_wallet.cpp]: fixed, takes sort + unique + erase

File: core/api/wallet/wallet_api.hpp [hunk 5]
```diff
@@ -59,6 +63,7 @@ namespace fc::api {
     f(a.WalletDefaultAddress);
     f(a.WalletHas);
     f(a.WalletImport);
+    f(a.WalletList);
```
Comment (turuslan) [inline core/api/wallet/wallet_api.hpp:66]:
```suggestion
    f(a.WalletList);
    f(a.WalletDelete);
```

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**